crates/ruff_python_formatter/src/expression/expr_f_string.rs

```diff
@@ -20,8 +22,21 @@ impl NeedsParentheses for ExprFString {
     fn needs_parentheses(
         &self,
         _parent: AnyNodeRef,
-        _context: &PyFormatContext,
+        context: &PyFormatContext,
     ) -> OptionalParentheses {
-        OptionalParentheses::Multiline
+        if self.implicit_concatenated {
+            OptionalParentheses::Multiline
+        } else {
+            let text = &context.source()[self.range];
+
+            if memchr2(b'\n', b'\r', text.as_bytes()).is_none()
+                && text.len() > 4
+                && text.len() < context.options().line_width().value() as usize
```

Comment:
This is always going to indented if it's parenthesized, right? So could we subtract the indent size from `context.options().line_width().value()`?

Reply:
We could, but we don't have an indent size option yet (only available in the printer). Let me add it in a follow up PR.